sway/input/seat.c

```diff
@@ -448,6 +448,20 @@ void seat_set_focus_warp(struct sway_seat *seat,
 		return;
 	}
 
+	struct sway_container *last_workspace = last_focus;
+	if (last_workspace && last_workspace->type != C_WORKSPACE) {
+		last_workspace = container_parent(last_workspace, C_WORKSPACE);
+	}
+	struct sway_container *new_workspace = container;
+	if (new_workspace && new_workspace->type != C_WORKSPACE) {
+		new_workspace = container_parent(new_workspace, C_WORKSPACE);
+	}
+
```

Comment:
Would you consider changing `last_output` and `new_output` init a few lines below? Now that we have workspaces defined, these are just  one `new_output = container_parent(new_workspace, C_OUTPUT)` away.

Reply:
I tried this just now, but `last_workspace` can be null which means I'd have to make it conditional. It doesn't really improve the readability in that case.

Reply:
I'd then settle on initially assigning `new_workspace` to `new_output`, `last_workspace` to `last_output` and leaving the `container_parent` calls inside if clauses.